### PR TITLE
Add initial Appstream metainfo

### PR DESCRIPTION
**Pull request by Newbytee**
Necessary for distributing on e.g. Flathub, but also useful for other means of distribution as it provides useful metadata for desktop environments and store applications. It should probably be installed by the CMake install script. Currently missing a releases tag since there are no releases. This would be necessary to publish on Flathub since they mandate that there is a releases tag, but it doesn't make sense to add one yet since there are no releases.

See https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Please review the values I put, especially the ID, summary, and metadata licence. GPL-2.0 is not a valid metadata licence unfortunately (see https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-metadata_license).

### CODE REVIEW

**Comment by Noordfrees** [comment]
Great :) 
A few comments:

```
$ appstreamcli validate data/packaging/com.github.freerct.FreeRCT.appdata.xml
E: com.github.freerct.FreeRCT:~: metadata-license-missing
I: com.github.freerct.FreeRCT:39: unknown-tag metadata_licence
I: com.github.freerct.FreeRCT:38: unknown-tag project_licence

Validation failed: errors: 1, infos: 2, pedantic: 1
```

I'd suggest calling the ID `org.freerct.FreeRCT`. The [recommendation](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic) is `{tld}.{vendor}.{product}`, and `org` is more common than `com` in FOSS games.

**Comment by Noordfrees** [comment]
Re `metadata_license` perhaps use `GFDL-1.2` – that's the *GNU Free Documentation License v1.2 or later* which is sort of the equivalent of the GPL-2 in the world of documentation.

**Comment by Newbytee** [comment]
> Great :) A few comments:
> 
> ```
> $ appstreamcli validate data/packaging/com.github.freerct.FreeRCT.appdata.xml
> E: com.github.freerct.FreeRCT:~: metadata-license-missing
> I: com.github.freerct.FreeRCT:39: unknown-tag metadata_licence
> I: com.github.freerct.FreeRCT:38: unknown-tag project_licence
> 
> Validation failed: errors: 1, infos: 2, pedantic: 1
> ```

I used appstream-util rather than appstreamcli. I'm not sure what the difference is between the two, but I can change it to use US spelling.

> I'd suggest calling the ID `org.freerct.FreeRCT`. The [recommendation](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic) is `{tld}.{vendor}.{product}`, and `org` is more common than `com` in FOSS games.

The issue with that is that you don't own the freerct.org domain and that it's in the hands of some scummy advertiser.

> Re metadata_license perhaps use GFDL-1.2 – that's the GNU Free Documentation License v1.2 or later which is sort of the equivalent of the GPL-2 in the world of documentation.

Sounds good to me, I'll change it later.

**Comment by Alberth289346** [comment]
Not sure why `com.freerct` would be much better, as scummy advertisers can buy that too if they haven't already.

What's recommended if you don't have a domain? Does it have to be a tld?, eg could `simgames` work, or `rct`?

In the end I don't care much about the ID, id isn't the same as an uri/url.

EDIT: `foss` ?

**Comment by Newbytee** [comment]
> Not sure why `com.freerct` would be much better, as scummy advertisers can buy that too if they haven't already.

It's not com.freerct. It's com.github.freerct.FreeRCT.

> What's recommended if you don't have a domain? Does it have to be a tld?, eg could `simgames` work, or `rct`?

Usually people do something like I did where they use GitHub or whatever they host on.

> In the end I don't care much about the ID, id isn't the same as an uri/url.

See this: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic

> The ID must follow a reverse-DNS scheme, consisting of {tld}.{vendor}.{product}, for example org.kde.gwenview or com.hugski.colorhug2. Ownership of {vendor}.{tld} in the domain name system guarantees uniqueness of IDs. 

> EDIT: foss ?

What is this in reference to?

**Comment by Noordfrees** [comment]
We don't know that FreeRCT will stay on GitHub forever, long-lived projects tend to move around a lot when better platforms appear. Having "GitHub" in the appdata ID far in the future might be a legacy burden, and this can not be easily changed once we've started releasing our first flatpak builds.

I don't see how using an owned domain would prevent duplication though, is this moderated somewhere or couldn't just anyone make a package and call it `com.github.freerct.FreeRCT.ScammyScammityScam`? 

I also wouldn't ever associate appdata IDs with actual URLs that can be typed into the browser. Personally I associate `com` with "commercial" (as in proprietary stuff) and `org` as the typical tld for **F**ree **O**pen **S**ource **S**oftware. (i.e., if I see a package that starts with `com.` I don't install it.) not sure if there are other common tld's that immediately signal "free-as-in-freedom package here"

**Comment by Alberth289346** [comment]
`.net` ?

If not, `org` seems indeed the better option despite not owning the domain.

**Comment by Noordfrees** [comment]
On a semi-related note, I'd be willing to rent a domain and virtual host for our own webserver in two months or thereabout (I cannot guarantee yet that I'll really be able to get around to it then though). `freerct.net` is currently [available](https://www.domain.com/registration/?flow=domainDFE&search=freerct#/domainDFE/1) and would be my first preference for our URL as freerct.org is taken. If having appdata ID and URL match is important perhaps we should put this PR on hold until we actually have a new domain; otherwise +1 for `net` or `org`.

**Comment by Newbytee** [comment]
It's not my opinion that it should follow a reverse-DNS scheme. The specification says that you must use that. If you get that domain you can use that, but using someone else's domain without permission is against the specification.

> I don't see how using an owned domain would prevent duplication though, is this moderated somewhere or couldn't just anyone make a package and call it com.github.freerct.FreeRCT.ScammyScammityScam?

It doesn't protect against malice, no. But it does protect against collisions as long as people are nice.

**Comment by Newbytee** [comment]
Btw, regarding hosting, Oracle (scummy company yes) provides free VPSes\: https://www.oracle.com/cloud/free/

**Comment by Newbytee** [comment]
I'll mark this as a draft until we've resolved the ID issue.

**Comment by Noordfrees** [comment]
I have registered the [freerct.net](http://freerct.net) domain now :) I'll start setting up a webserver over the next days, until then it redirects to our GitHub-hosted website.

---

Edit: The **repository** is at https://github.com/Noordfrees/FreeRCT-website. 

Another edit: The website is up at https://freerct.net now. I'm stills setting up lots of essential functionality and will report when it's ready. Gimme another couple of days…

**Comment by Newbytee** [comment]
Hey again, I've updated the PR. It includes some other Flatpak-related changes now as I felt it made sense to bundle those with this, so please review that. I imagine this may break the Debian package as I moved the desktop file, but I don't know how Debian packaging works so I'd like it if you could look at that too.

**Comment by Newbytee** [comment]
Here's what it looks like in GNOME Software with this metadata by the way:

image

image

**Comment by fnetX** [comment]
I'd replace the screenshot at the right tbh, maybe the one at the left, too. The UI for the action popovers doesn't look very inviting, and the empty one might look better with at least a started construction.

**Comment by Alberth289346** [comment]
Perhaps drop the less interesting screenshots

**Comment by Newbytee** [comment]
Not sure I understand why the Windows x64 pipeline failed. It doesn't seem to have much in the way of information. Could I re-run it somehow?

**Comment by Newbytee** [comment]
> Perhaps drop the less interesting screenshots

From the repo entirely or just from the metainfo?

**Comment by Newbytee** [comment]
I removed the last two screenshots.

**Comment by Newbytee** [comment]
So now there are just these two:

image

File: packaging_data/freerct.metainfo.xml.in [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	<launchable type="desktop-id">@FREERCT_BASENAME@.desktop</launchable>
+	<screenshots>
+		<screenshot type="default">
+			<caption>A park in the game</caption>
+			<image>https://freerct.github.io/FreeRCT/images/screenshots/0_1/mainview.png</image>
+		</screenshot>
+		<screenshot>
+			<caption>The game's main menu</caption>
+			<image>https://freerct.github.io/FreeRCT/images/screenshots/0_1/mainmenu.png</image>
+		</screenshot>
```
Comment (Noordfrees) [inline packaging_data/freerct.metainfo.xml.in]:
```suggestion
		<screenshot type="default">
			<caption>A park in the game</caption>
			<image>https://freerct.net/img/screenshots/appdata/mainview.png</image>
		</screenshot>
		<screenshot>
			<caption>The game's main menu</caption>
			<image>https://freerct.net/img/screenshots/appdata/mainmenu.png</image>
		</screenshot>
```

**Review by Noordfrees — APPROVED** [review]
Perfect :)

**Pull request merged**